dg-spark/src/main/code/org/finra/datagenerator/SparkDistributor.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2014 DataGenerator Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.finra.datagenerator
+
+import java._
+import java.io.{ObjectOutputStream, OutputStream}
+
+import org.finra.datagenerator.consumer.{DataPipe, DataConsumer}
+import org.finra.datagenerator.writer.{DefaultWriter, DataWriter}
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.finra.datagenerator.distributor.SearchDistributor
+import org.finra.datagenerator.engine.Frontier
+
+/**
+ * Take Frontiers produced by an Engine and process the Frontiers in parallel,
+ * giving the results to a DataConsumer for post-processing.
+ *
+ * Set masterURL to SparkConf using Parametrized Constructor
+ *
+ * Created by Brijesh on 6/2/2015.
+ */
+class SparkDistributor(masterURL: String) extends SearchDistributor with java.io.Serializable {
+
+  val flag: Boolean = true
+
+  val javaQueue = new util.LinkedList[util.Map[String, String]]()
+
+  //Define Spark Context and Data Consumer
+  var sparkContext: SparkContext = null
```

Comment:
Your scalastyle will give warnings on these lines because you're using null. And it's a good rule if you can avoid it -- perhaps if you can define them through the constructor.